### ISSUE TITLE
FR: compatibility with OpenAI's API

### ISSUE THREAD

**Issue by didmar** [ahmetoner/whisper-asr-webservice#182]
If I'm not mistaken, this API is different from OpenAI's.
It prevents usage as a drop-in replacement for other applications or plugins that connect to Whisper (such as https://github.com/nikdanilov/whisper-obsidian-plugin).
This would be great to avoid ad-hoc implementations in each of these applications and plugins.

**Comment by ayancey** [comment]
I like this idea. It would be useful if you could explain OpenAI's endpoints a bit and what the output format is for the data. thank you

**Comment by didmar** [comment]
I've ended up forking a project that does specifically that: https://github.com/didmar/whisper-api-server
There is also the official doc: https://platform.openai.com/docs/guides/speech-to-text

**Comment by ahmetoner** [comment]
Good idea! I was actually planning to implement this change in version 2.0. I'll start to make some breaking changes to the repository after the new year.

**Comment by fedirz** [comment]
I've recently created [faster-whisper-server](https://github.com/fedirz/faster-whisper-server), which supports `/v1/audio/transcriptions`, `/v1/audio/transcriptions`, `/v1/models`, and `/v1/models/{model}` routes, as well as streaming transcription/translation (SSE), and live audio transcription (web socket)

**Comment by ayancey** [comment]
proof of concept: #237. Still needs a lot of love and some abstraction to work across openai-whisper/faster-whisper cores
